crates/services/executor/src/executor.rs

```diff
@@ -644,11 +728,27 @@ where
                     *coinbase_contract_id,
                     memory,
                 ) {
-                    Ok(_) => {}
+                    Ok(_) => {
+                        // SAFETY: Transaction succeed so we know that `tx_status` has been populated
+                        let latest_executed_tx = data.tx_status.last().expect(
+                            "Shouldn't happens as we just added a transaction; qed",
+                        );
+                        let preconfirmation_status =
+                            convert_tx_execution_result_to_preconfirmation(
+                                &latest_executed_tx.result,
+                                *block.header.height(),
+                                data.tx_count,
+                            );
+                        status.push(preconfirmation_status);
+                    }
                     Err(err) => {
+                        status.push(PreconfirmationStatus::SqueezedOutByBlockProducer {
```

Comment:
Should this be:
```rs
    /// Transaction will not be included in a block, rejected at `block_height`
    FailureByBlockProducer {
        /// The block height at which the transaction will be rejected
        block_height: BlockHeight,
    },
 ```
 instead of squeezed out? I thought squeezing out happens in the txpool.

Reply:
No, Failed transaction will be part of the `Ok(result)` branch. If we got an error during execution on executor level, it means transaction was invalidate from the begging, and we shouldn't include it in the block at all(Like uses non existing inputs, or fails basic validation rules).